pandas/core/indexes/datetimes.py

```diff
@@ -906,25 +905,13 @@ def _format_native_types(self, na_rep='NaT', date_format=None, **kwargs):
     @Appender(_index_shared_docs['astype'])
     def astype(self, dtype, copy=True):
         dtype = pandas_dtype(dtype)
-        if is_object_dtype(dtype):
-            return self._box_values_as_index()
-        elif is_integer_dtype(dtype):
-            return Index(self.values.astype('i8', copy=copy), name=self.name,
-                         dtype='i8')
-        elif is_datetime64_ns_dtype(dtype):
+        if is_datetime64_ns_dtype(dtype):
```

Comment:
can it be merged to `datetimelike` using `is_dtype_equal(self.dtype, dtype)`?

Reply:
Nice catch.  I think it can be merged all the way down to `base`.